File: Books_Ecom_Project_171_Models/Company.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Books_Ecom_Project_171_Models
{
    public class Company
    {
        public int Id { get; set; }
        public string Name { get; set; }
        [Display(Name = "Street Address")]
        public string StreetAddress { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        [Display(Name = "Postal Code")]
        public string PostalCode { get; set; }
        [Display(Name = "Phone Number")]
        public string PhoneNumber { get; set; }
        [Display(Name = "Is Authorized Company")]
        public bool IsAuthorizedCompany { get; set; }
    }
}

File: Books_Ecom_Project_171/wwwroot/js/company.js
var dataTable;

$(document).ready(function () {
    loadDataTable()
})
function loadDataTable() {
    dataTable = $('#tblData').DataTable({
        "ajax": {
            "url": "/Admin/Company/GetAll",

        },
        "columns": [
            { "data": "name", "width": "12%" },
            { "data": "streetAddress", "width": "12%" },
            { "data": "city", "width": "12%" },
            { "data": "state", "width": "12%" },
            { "data": "postalCode", "width": "12%" },
            { "data": "phoneNumber", "width": "12%" },
            {
                "data": "isAuthorizedCompany",
                "width": "12%",
                "render": function (data) {
                    if (data) {
                        return `
                        <input type="checkbox" checked disable/>
                        `;
                    }
                    else {
                        return `
                        <input type="checkbox" checked/>
                        `;
                    }
                }
            },
            {
                "data": "id",
                "render": function (data) {
                    return `
                    <div class="text-center">
                    <a href="/Admin/Company/Upsert/${data} " class="btn btn-primary" >
                    <i class="fas fa-edit"></i>
                    </a>
                    <a class="btn btn-danger" onclick=Delete('/Admin/Company/Delete/${data}')>
                    <i class="fas fa-trash"></i>
                    </a>
                    </div>
                    `;
                }
            }
        ]
    })
}
function Delete(url) {
    alert(url);
}